semantic/src/main/kotlin/tools/samt/semantic/Types.kt

```diff
@@ -116,6 +116,10 @@ sealed interface UserDeclared {
     val declaration: Node
 }
 
+sealed interface Annotated {
+    val declaration: AnnotatedNode
+}
```

Comment:
You could add a convenience property for accessing the annotations if that's something you think will make the code easier to work with
```suggestion
sealed interface Annotated {
    val declaration: AnnotatedNode
    val annotations get() = declaration.annotations
}
```